src/hooks/useFileUpload.ts

```diff
@@ -0,0 +1,42 @@
+import { useState, useRef, ChangeEvent } from "react";
+
+export const useFileUpload = () => {
+  const fileInputRef = useRef<HTMLInputElement | null>(null);
```

Comment:
const fileInputRef = useRef<HTMLInputElement | null>(null); this would be in inputUpload component